src/import/sst_importer.rs

```diff
@@ -0,0 +1,386 @@
+        })
+    }
+
+    fn join(&self, meta: &SSTMeta) -> Result<ImportPath> {
+        let file_name = sst_meta_to_path(meta)?;
+        let save_path = self.root_dir.join(&file_name);
+        let temp_path = self.temp_dir.join(&file_name);
+        let clone_path = self.clone_dir.join(&file_name);
+        Ok(ImportPath {
+            save: save_path,
+            temp: temp_path,
+            clone: clone_path,
+        })
+    }
+
+    fn create(&self, meta: &SSTMeta) -> Result<ImportFile> {
+        let path = self.join(meta)?;
+        if path.save.exists() {
+            return Err(Error::FileExists(path.save));
+        }
+        ImportFile::create(meta.clone(), path)
+    }
+
+    fn ingest(&self, meta: &SSTMeta, db: &DB) -> Result<()> {
+        let path = self.join(meta)?;
+        let cf = meta.get_cf_name();
+        prepare_sst_for_ingestion(&path.save, &path.clone)?;
+        validate_sst_for_ingestion(db, cf, &path.clone, meta.get_length(), meta.get_crc32())?;
+
+        let handle = get_cf_handle(db, cf)?;
+        let mut opts = IngestExternalFileOptions::new();
+        opts.move_files(true);
+        db.ingest_external_file_cf(handle, &opts, &[path.clone.to_str().unwrap()])?;
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct ImportPath {
+    // The path of the file that has been uploaded.
+    save: PathBuf,
+    // The path of the file that is being uploaded.
+    temp: PathBuf,
+    // The path of the file that is going to be ingested.
+    clone: PathBuf,
+}
+
+impl fmt::Debug for ImportPath {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("ImportPath")
+            .field("save", &self.save)
+            .field("temp", &self.temp)
+            .field("clone", &self.clone)
+            .finish()
+    }
+}
+
+pub struct ImportFile {
+    meta: SSTMeta,
+    path: ImportPath,
+    file: Option<File>,
+    digest: crc32::Digest,
+}
+
+impl ImportFile {
+    fn create(meta: SSTMeta, path: ImportPath) -> Result<ImportFile> {
+        let file = OpenOptions::new()
+            .write(true)
+            .create_new(true)
+            .open(&path.temp)?;
+        Ok(ImportFile {
+            meta: meta,
+            path: path,
+            file: Some(file),
+            digest: crc32::Digest::new(crc32::IEEE),
+        })
+    }
+
+    fn append(&mut self, data: &[u8]) -> Result<()> {
+        self.file.as_mut().unwrap().write_all(data)?;
+        self.digest.write(data);
+        Ok(())
+    }
+
+    fn finish(&mut self) -> Result<()> {
+        self.validate()?;
+        self.file.take().unwrap().sync_all()?;
+        if self.path.save.exists() {
+            return Err(Error::FileExists(self.path.save.clone()));
+        }
+        fs::rename(&self.path.temp, &self.path.save)?;
+        Ok(())
+    }
+
+    fn cleanup(&mut self) -> Result<()> {
+        self.file.take();
+        if self.path.temp.exists() {
+            fs::remove_file(&self.path.temp)?;
+        }
```

Comment:
How about if `temp file` have been renamed to `save path`?

Reply:
`cleanup` is called when this file is dropped:
1. if the file is finished, the temp file has been renamed to save path, we have nothing to do.
2. if the file is not finished, the temp file will not be used anymore, we can remove it.
So, in both cases, if the temp file exists when the file is dropped, we can just remove it.